pkg/kubecost/window.go

```diff
@@ -456,7 +456,7 @@ func (w Window) Hours() float64 {
 }
 
 func (w Window) IsEmpty() bool {
-	return !w.IsOpen() && w.end.Equal(*w.Start())
+	return w.start == nil && w.end == nil
```

Comment:
My original take here was that the previous impl was a typo and should be `return w.IsOpen() && w.end.Equal(*w.Start())`. If felt that `return w.start == nil && w.end == nil` conveyed that more succinctly but reading it again it appears that the original author meant that a window that covers no-time is empty. I feel that IsEmpty is a better idea for what I was trying to express. And a separate function like `window.HasDurration()` makes more sense.

Reply:
I'm still a bit worried about changing the definition of `IsEmpty()` here. I think the intent of a `nil` start or end is to represent an unbounded/infinite side of a window.

E.g.
- (start, end) = (now, nil) represents [now, ∞)
- (start, end) = (nil, now) represents (∞, now]
- (start, end) = (nil, nil) represents (∞, ∞)

I don't think it's quite right to say that (∞, ∞) is an "empty" window. It might be better to name the current `IsEmpty()` functionality as something more like `HasFiniteNonZeroDuration()`, but that's kind of weird, too.

If you need a method to check `w.start == nil && w.end == nil`, how about a new method like `func (w Window) IsFullyOpen() bool`?